src/core/src/CoreMain.py

```diff
@@ -74,10 +74,10 @@ def __init__(self, argv):
             if not execution_config.exec_auto_assess_only:
                 configure_patching_successful = configure_patching_processor.start_configure_patching()
 
-            # Assessment happens for an Auto Assessment request or for all Non Auto Assessment operations, except for ConfigurePatching iff AssessmentMode is set to AutomaticByPlatform
-            include_assessment_with_configure_patching = (patch_operation_requested == Constants.CONFIGURE_PATCHING.lower() and execution_config.assessment_mode == Constants.AssessmentModes.AUTOMATIC_BY_PLATFORM)
-            if execution_config.exec_auto_assess_only or patch_operation_requested != Constants.CONFIGURE_PATCHING.lower() or include_assessment_with_configure_patching:
-                patch_assessment_successful = patch_assessor.start_assessment()
+            # Assessment happens for all operations. If the goal seeking tracked operation is CP, then its final status can only be written after assessment reaches a terminal state
+            patch_assessment_successful = patch_assessor.start_assessment()
+            if not execution_config.exec_auto_assess_only and patch_operation_requested == Constants.CONFIGURE_PATCHING.lower():
+                configure_patching_processor.set_configure_patching_final_overall_status()
```

Comment:
If assessment throws (raises?) an exception, does the except ensure a final status is written?

Reply:
Assessment has built in full-cover exception handling with retries, and there's status reconciliation in the highest-level exception handling, but there's still a slim possibility of a super-fundamental problem (like being unable to log something basic that is never expected to fail and would need our attention anyway) that could throw this off. Evaluating if something additional should be done for that.

Reply:
Addressed with more safety checks.